src/context/LogContext/index.tsx

```diff
@@ -82,15 +89,29 @@ const LogContextProvider: React.FC<LogContextProviderProps> = ({
 
   // TODO EVG-17537: more advanced filtering
   const processedLogLines = useMemo(
-    () => filterLogs(state.logs, filters, bookmarks, selectedLine, filterLogic),
+    () =>
+      filterLogs({
+        logLines: state.logs,
+        filters,
+        bookmarks,
+        selectedLine,
+        filterLogic,
+        expandedLines: state.expandedLines,
+        expandableRows,
+      }),
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [state.logs.length, `${filters}`, `${bookmarks}`, selectedLine, filterLogic]
+    [
+      state.logs.length,
+      `${filters}`, // eslint-disable-line react-hooks/exhaustive-deps
+      `${bookmarks}`, // eslint-disable-line react-hooks/exhaustive-deps
+      `${state.expandedLines}`, // eslint-disable-line react-hooks/exhaustive-deps
+      selectedLine,
+      filterLogic,
+      expandableRows,
+    ]
   );
```

Comment:
We can use the new [`useDeferredValue`](https://reactjs.org/docs/hooks-reference.html#usedeferredvalue) hook to defer this state update. This function is the slowest part of the context and is what causes the app to lag when rendering it. We are already memoizing its value to prevent it from rerendering but we can also defer its updates when it does run to avoid blocking the UI. In my experience this lead to a pretty dramatic improvement in performance. 
```ts
  const deferredFilters = useDeferredValue(filters);
  
  // TODO EVG-17537: more advanced filtering
  const processedLogLines = useMemo(
    () => {
      return filterLogs({
        logLines: state.logs,
        filters,
        bookmarks,
        selectedLine,
        filterLogic,
        expandedLines: state.expandedLines,
        expandableRows,
      });
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [
      state.logs.length,
      `${deferredFilters}`, // eslint-disable-line react-hooks/exhaustive-deps
      `${bookmarks}`, // eslint-disable-line react-hooks/exhaustive-deps
      `${state.expandedLines}`, // eslint-disable-line react-hooks/exhaustive-deps
      selectedLine,
      filterLogic,
      expandableRows,
    ]
  );



```